hi-ml/src/health_ml/runner.py

```diff
@@ -0,0 +1,299 @@
+
+        self.experiment_config = experiment_config
+        if not experiment_config.model:
+            raise ValueError("Parameter 'model' needs to be set to specify which model to run.")
+        print(f"Creating model loader with the following args: {parser_result.args}")
+        model_config_loader: ModelConfigLoader = ModelConfigLoader(**parser_result.args)
+        # Create the model as per the "model" commandline option. This is a LightningContainer.
+        container = model_config_loader.create_model_config_from_name(model_name=experiment_config.model)
+
+        # parse overrides and apply
+        assert isinstance(container, param.Parameterized)
+        parser_ = create_argparser(container)
+        # For each parser, feed in the unknown settings from the previous parser. All commandline args should
+        # be consumed by name, hence fail if there is something that is still unknown.
+        parser_result_ = parse_arguments(parser_, args=parser_result.unknown)
+        # Apply the overrides and validate. Overrides can come from either YAML settings or the commandline.
+        _ = apply_overrides(container, parser_result_.overrides)  # type: ignore
+        container.validate()
+
+        self.lightning_container = container
+
+        return parser_result_
+
+    def run(self) -> Tuple[LightningContainer, AzureRunInfo]:
+        """
+        The main entry point for training and testing models from the commandline. This chooses a model to train
+        via a commandline argument, runs training or testing, and writes all required info to disk and logs.
+
+        :return: a tuple of the LightningContainer object and an AzureRunInfo containing all information about
+            the present run (whether running in AzureML or not)
+        """
+        # Usually, when we set logging to DEBUG, we want diagnostics about the model
+        # build itself, but not the tons of debug information that AzureML submissions create.
+        logging_to_stdout(logging.INFO if is_local_rank_zero() else "ERROR")
+        initialize_rpdb()
+        self.parse_and_load_model()
+        azure_run_info = self.submit_to_azureml_if_needed()
+        self.run_in_situ(azure_run_info)
+        return self.lightning_container, azure_run_info
+
+    def submit_to_azureml_if_needed(self) -> AzureRunInfo:
+        """
+        Submit a job to AzureML, returning the resulting Run object, or exiting if we were asked to wait for
+        completion and the Run did not succeed.
+
+        :return: an AzureRunInfo object containing all of the details of the present run. If AzureML is not
+            specified, the attribute 'run' will None, but the object still contains helpful information
+            about datasets etc
+        """
+        root_folder = self.project_root
+        entry_script = Path(sys.argv[0]).resolve()
+        script_params = sys.argv[1:]
+
+        additional_conda_env_files = self.lightning_container.additional_env_files
```

Comment:
We may need to re-think that. Should the additional environment file be picked up automagically by default? Say someone has a repository that uses hi-ml, they'd have a conda environment specified there, and their models. So it would make sense to search for environment.yml in the current working directory and upwards (assuming that the cwd is in the repo). If there's nothing to be found there, yes, get it from the container.

Reply:
It picks up on the working directory environment.yml file already in `get_all_environment_files`. But there may be scenarios in which a user wants to specify yet more env files - e.g. currently if running one of the histo models in innereye-deeplearning, you'd need to specify innereye-deeplearning/environment.yml as an additional env file. If we searched for all env files from the root (i.e. working directory, i.e. folder that contains innereye-deeplearning) level down, we could pick up on unecesary env files that could potentially lead to version conflicts. Thoughts?